core/math/a_star_grid_2d.cpp

```diff
@@ -108,14 +108,33 @@ Size2 AStarGrid2D::get_cell_size() const {
 	return cell_size;
 }
 
+void AStarGrid2D::set_tile_shape(TileShape p_tile_shape) {
+	if (tile_shape != p_tile_shape) {
+		ERR_FAIL_INDEX(p_tile_shape, TileShape::TILE_SHAPE_MAX);
+		tile_shape = p_tile_shape;
+		dirty = true;
+	}
```

Comment:
```suggestion
	if (tile_shape == p_tile_shape) {
		return;
	}

	ERR_FAIL_INDEX(p_tile_shape, TileShape::TILE_SHAPE_MAX);
	tile_shape = p_tile_shape;
	dirty = true;
```
I think early exit is cleaner and is generally the pattern